draft-ietf-quic-transport.md

```diff
@@ -1896,6 +1896,13 @@ dynamically calculate the expiration time.  It is also unlikely that the client
 port number is the same on two different connections; validating the port is
 therefore unlikely to be successful.
 
+Even when a token is valid, a server SHOULD limit the amount of data that it
+sends toward a client in response to 0-RTT data.  Spoofing of source addresses
+and replays of 0-RTT handshake messages can cause tokens to appear to be valid
+when they are not.  No specific recommendation is made regarding the ratio of
+data sent to received, though it is unlikely to be necessary to send more than
+the initial congestion window.
```

Comment:
This text seems wrong. Tokens are either valid or they are not.

What specific attack scenario are you concerned with?

Reply:
Tokens might be ignored (as we discussed, they might be old, or the keys might not be available).

This is your issue #1309.

Reply:
I think it's fairly likely it may be necessary to send more than the IW, but I think we should advise limiting the amount of data sent to the IW.  I think this just falls out of the recovery doc, since the recovery doc specifies an initial window of 10 packets, so I'm not completely sure this is necessary?

Reply:
Yeah, this is why it is worded so loosely.  Without this advice, the risk seems to be limited to the server receiving a spoofed ACK for the ServerHello and sending that much more in response, but that's only one more packet than the IW, so it's hardly earth-shattering.

Given that this is largely dictated by congestion controller design (IW specifically), is this really a change we need to implement?  Should we instead close #1309 with no change?

Reply:
I'm fine with just closing #1309

Reply:
@ekr?